rust/operator-binary/src/crd/mod.rs

```diff
@@ -816,16 +833,26 @@ impl v1alpha1::TrinoCluster {
             .collect::<BTreeMap<_, _>>()
             .into_iter()
             .flat_map(move |(rolegroup_name, rolegroup)| {
-                let rolegroup_ref = TrinoRole::Coordinator.rolegroup_ref(self, rolegroup_name);
+                let role_group_ref = TrinoRole::Coordinator.rolegroup_ref(self, rolegroup_name);
                 let ns = ns.clone();
                 (0..rolegroup.replicas.unwrap_or(0)).map(move |i| TrinoPodRef {
                     namespace: ns.clone(),
-                    role_group_service_name: rolegroup_ref.object_name(),
-                    pod_name: format!("{}-{}", rolegroup_ref.object_name(), i),
+                    role_group_service_name: Self::rolegroup_metrics_service_name(
+                        &role_group_ref.object_name(),
+                    ),
+                    pod_name: format!(
+                        "{role_group}-{i}",
+                        role_group = role_group_ref.object_name()
+                    ),
                 })
             }))
     }
 
+    /// Returns the metrics rolegroup service name `simple-trino-coordinator-default-<METRICS_SERVICE_SUFFIX>`.
```

Comment:
Do we need this attached to the cluster, instead of being a stand-alone function?